app/commands.py

```diff
@@ -204,6 +243,134 @@ def sync_opensearch(start_page=1, per_page=100, recreate_index=False):
         raise click.ClickException(f"Sync failed: {type(e).__name__}")
 
 
+@search.cli.command("compare")
+@click.option(
+    "--sample-size",
+    default=10,
+    show_default=True,
+    help="How many example IDs to print for each discrepancy type.",
+)
+@click.option(
+    "--fix",
+    is_flag=True,
+    help="Automatically index missing datasets and delete extra docs from OpenSearch.",
+)
+def compare_opensearch(sample_size: int, fix: bool):
+    """Report (and optionally fix) dataset ID discrepancies between DB and OpenSearch."""
+
+    interface = CatalogDBInterface()
+    client = OpenSearchInterface.from_environment()
+
+    click.echo("Collecting dataset IDs from DB…")
+    db_ids = {row[0] for row in interface.db.query(Dataset.id).all()}
+    click.echo(f"Database datasets: {len(db_ids)}")
+
+    click.echo("Collecting document IDs from OpenSearch…")
+    os_ids = {
+        hit["_id"]
+        for hit in scan(client.client, index=client.INDEX_NAME, _source=False)
+    }
+    click.echo(f"OpenSearch documents: {len(os_ids)}")
```

Comment:
I'd love to see a test for this, but don't want to hold this up. Don't see any problems with it.